server/perfkit/common/gae_cloud_sql_client.py

```diff
@@ -0,0 +1,155 @@
+"""Copyright 2016 Google Inc. All rights reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+
+A client for accessing Cloud SQL from appengine.
+"""
+
+__author__ = 'klausw@google.com (Klaus Weidner)'
+
+import MySQLdb
+import datetime
+import logging
+import os
+
+from perfkit.common import big_query_client
+
+# Setting up readonly user rights, assumes the 'readonly' user already exists:
+#
+# mysql> revoke all privileges, grant option from readonly@localhost;
+# mysql> grant select on mystuff.* to readonly@localhost;
+# mysql> show grants for readonly@localhost;
+#------------------------------------------------------------------------------+
+# Grants for readonly@localhost                                                |
+#------------------------------------------------------------------------------+
+# GRANT USAGE ON *.* TO 'readonly'@'localhost' IDENTIFIED BY PASSWORD <secret> |
+# GRANT SELECT ON `mystuff`.* TO 'readonly'@'localhost'                        |
+#------------------------------------------------------------------------------+
+
+# TODO(klausw): The password isn't needed according to docs, but I can't get it
+# working without it. A hardcoded password seems ok in this case since access
+# is limited to 'localhost' and also restricted by appengine application ID.
```

Comment:
Per discussion, will add username and password fields to the client-side config/JSON so that these aren't hardcoded.  Will also plan for a future feature of "predefined datasources" so that there is a central registry of instance/db/username/password entries.